plugins/rbac-backend/src/role-manager/role-manager.ts

```diff
@@ -11,13 +12,17 @@ import { RoleList } from './role-list';
 
 export class BackstageRoleManager implements RoleManager {
   private allRoles: Map<string, RoleList>;
+  private maxDepth?: number;
   constructor(
     private readonly catalogApi: CatalogApi,
     private readonly log: Logger,
     private readonly tokenManager: TokenManager,
     private readonly catalogDBClient: Knex,
+    private readonly config: Config,
   ) {
     this.allRoles = new Map<string, RoleList>();
+    const rbacConfig = this.config.getOptionalConfig('permission.rbac');
+    this.maxDepth = rbacConfig?.getOptionalNumber('maxDepth');
```

Comment:
It would be nice to validate this value somewhere. It should be positive, like you described in the doc.